dcp-0006/dcp-0006.mediawiki

```diff
@@ -0,0 +1,347 @@
+<pre>
+DCP: 0006
+Title: Decentralized Treasury
+Author: Marco Peereboom <marco@decred.org>
+Status: Defined
+Created: 2020-09-18
+License: CC0-1.0
+License-Code: ISC
+</pre>
+
+==Abstract==
+
+Spending from the Decred treasury is a manual process that is executed and
+governed by humans. The current process requires contractors to each produce a
+monthly invoice with a reimbursement report, denominated in USD, that is
+submitted to the project treasury auditors. If the invoice is approved a payout
+is made to the contractor-provided address from the treasury funds, using an
+averaged DCR/USD exchange rate over the month the invoice applies to. These
+payouts are made by Decred Holdings Group LLC ("DHG" for short), a
+conventional corporate entity which holds the treasury funds.
+
+This specifies modifications to the Decred treasury to make it a more
+decentralized process.
+
+The main changes are:
+
+* Move multisig utxo based treasury address to a treasury account that can only be debited when Decred stakeholders agree to it.
+* Credit development subsidy to treasury account every block.
+* Add three new opcodes: <code>OP_TADD</code>, <code>OP_TSPEND</code>, and <code>OP_TGEN</code>.
+* The aforementioned opcodes are then in turn used to construct the three new types of stake transactions:
+*# Treasurybase transaction - treasury subsidy in every block.
+*# Treasury add transactions - method to credit the treasury account.
+*# Treasury spend transactions - method to debit the treasury account.
+
+==Motivation==
+
+At this present time, all Decred treasury payouts are handled by DHG. This is
+undesirable for various reasons, such as
+
+* Sovereignty over treasury spends is not vested in the stakeholders.
+* Risk of theft of funds.
+* Risk of key project members abandoning the project.
+* Risk of coercion being used against key project members.
+* Process of treasury spending is insufficiently transparent.
+
+The proposed solution will semi-automate the release of treasury funds while
+diminishing the influence of key project members over the treasury spends.
+
+==Process==
+
+Currently, every block that is mined must include a coinbase transaction that
+generates a utxo of a specific amount that requires a 2-of-3 multisig to spend.
+This is enforced by consensus. During payout time, DHG creates and signs
+transactions to all payees. This process is not adequate to support the
+features proposed in this DCP.
+
+Essentially, the treasury is converted from a utxo based model to an account
+based model. The account balance is determined by starting with an initial
+balance of zero followed by state changes that are recorded on the blockchain.
+This process "burns" and "creates" utxos as needed. Sending utxos to the
+treasury burns the coins and adds value to the treasury account and spending
+from the treasury creates utxos.  This process is analogous to the ticket
+purchases and redeems. The treasury changes shall be added to the staking
```

Comment:
> The treasury changes shall be added to the staking mechanism.

Is it worth explaining **why** its being added to the staking mechanism? 

One of the effects of moving the transactions which add block subsidy to the treasury from the regular tx tree to the stake tx tree, is that these transactions can no longer be rejected if PoS voters disapprove the block which includes them.

I'm not sure if that was _**a reason**_ for choosing to move them into the stake tree, or if is merely _**a side-effect**_, but it may be worth calling out either way.